### ISSUE TITLE
[PHP 8] Login loop when using PHP > 7.4

### ISSUE THREAD

**Issue by Scindix** [phpmyadmin/phpmyadmin#16915]
### Describe the bug

When I'm using PHP 8 and try to login I'm immediately redirected to the same login page. Which means I cannot get past the login mask. It's important to note that the bug **does not occur** when I downgrade to **PHP 7.4**. So I guess this is a compatibility issue with PHP 8.

### To Reproduce

1. Install PHP 8
2. Try to login

### Expected behavior

After providing the correct login credentials I should be forwarded to the main page of phpMyAdmin.

### Screenshots

![pma_login_loop](https://user-images.githubusercontent.com/8752119/119420211-390f0c00-bcfc-11eb-8ad2-ead53550b5f3.gif)

### Server configuration

- Operating system: Arch Linux
```
$ uname -a && lsb_release -a
Linux andromeda 5.11.16-zen1-1-zen #1 ZEN SMP PREEMPT Wed, 21 Apr 2021 17:22:09 +0000 x86_64 GNU/Linux
LSB Version:	1.4
Distributor ID:	Arch
Description:	Arch Linux
Release:	rolling
Codename:	n/a
```
- Web server: Apache 2.4
```
$ apachectl -V
Server version: Apache/2.4.46 (Unix)
Server built:   Oct 14 2020 18:59:56
Server's Module Magic Number: 20120211:93
Server loaded:  APR 1.7.0, APR-UTIL 1.6.1
Compiled using: APR 1.7.0, APR-UTIL 1.6.1
Architecture:   64-bit
Server MPM:     prefork
  threaded:     no
    forked:     yes (variable process count)
```
- Database version: MariaDB 10.5
```
$ mysqld --version
mysqld  Ver 10.5.10-MariaDB for Linux on x86_64 (Arch Linux)
```
- PHP version: PHP 8.0
```
$ php --version
PHP 8.0.5 (cli) (built: Apr 27 2021 18:07:13) ( NTS )
```
- phpMyAdmin version: 5.1.0
Edit: As requested I also tested the latest 5.1 (and 5.2) snapshots. They exhibit the same behavior.
  - **phpMyAdmin 5.1+snapshot**: Daily snapshot, generated 2021-05-24, from Git commit 68c113b5594fef3cbb90f91fe9fbbfb383b4ac25.
  - **phpMyAdmin 5.2+snapshot**: Daily snapshot, generated 2021-05-24, from Git commit 06598d7b6b94f876560d092a6990d315cf43dac1
```
$ pacman -Qs phpmyadmin
local/phpmyadmin 5.1.0-1
```
- Since it might be relevant, here are the installed and activated PHP extensions:
```
$ php -m
[PHP Modules]
Core
ctype
curl
date
dom
fileinfo
filter
hash
iconv
json
libxml
mbstring
mysqli
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
pdo_pgsql
pdo_sqlite
pgsql
Phar
posix
readline
Reflection
session
SimpleXML
SPL
sqlite3
standard
tokenizer
xml
xmlreader
xmlrpc
xmlwriter
zip
zlib

[Zend Modules]
```

### Client configuration

- Browser: same results with Firefox 88.0, Gnome Web 40.1 and  Chromium 90.0
- Operating system: Arch Linux (same as server)

### Additional context

I first reported this issue on the Arch Linux Forum. When I realized this behaviour is related to the PHP version I figured that it might be a bug and thus I'm reporting it here now. I included all relevant information in this post so there is really no need. But just in case you are interested and as sort of a crosspost disclaimer, here is [my original post on the forum](https://bbs.archlinux.org/viewtopic.php?id=266649).

**Comment by williamdes** [comment]
Hi!

First off all, thank you for the detailed report!

It looks similar to #16544

Would you mind trying the latest 5.1 version in development [(phpMyAdmin 5.1+snapshot)](https://www.phpmyadmin.net/downloads/#snapshot_5.1+snapshot)?

**Comment by Scindix** [comment]
You're welcome! Also thanks for your fast answer.

I just tried. But unfortunately the bug still occurs with the current snapshot you linked to. I tested it with PHP 8 and PHP 7.4. It works with the latter but not the former - just like before. I then also tried the latest 5.2 version. But it exhibits the same behaviour.
I also updated the original bug report to reflect the additional tests. (Marked with "Edit:")

Is there any further debugging information I can provide? There seem to be no errors in my PHP log. Although I don't know how phpMyAdmin handles error reporting...

I probably should have mentioned that I also tried PHP 8 with my own projects. And they worked fine. Even when I was using mysqli to create some databases and tables. So it's most likely not a problem with my PHP 8 installation either.

**Comment by williamdes** [comment]
Well this is quite strange, I use PHP 8 & PHP 8.1 all day with phpMyAdmin and I never had this issue 🤔 

Is there some difference between the 7.4 and 8.0 installations ?

It looks like PHP never receives the data. What does it do if you login without entering any data ?
Does it detect if you enter a username without a password ?
Also are you using a proxy or HTTPS ?

**Comment by Scindix** [comment]
> Is there some difference between the 7.4 and 8.0 installations ?

Well actually there aren't that many differences. When I posted this issue I compared the extensions, but there wasn't anything noteworthy. For the rest of the php.ini I assumed they're basically the same since I didn't changed the preinstalled config files. Or so I thought. But when I ran `diff` on them and compared everything line by line I found my problem.

I **completely** forgot that I had to disable post data reading for a different project some time ago. (It's the only way to make PHP not replace dots with underscores for `multipart/form-data`). No wonder I had this issue. After reenabling it phpMyAdmin works again. I ended up only disabling it for my project's directory and everything works flawlessly now.

```apacheconf
<Directory "/srv/http/myproject">
    php_value enable_post_data_reading Off
</Directory>
```

Anyway, sorry for the inconvenience. But thanks for helping me in finding the issue.
